gecko/glue/nsObserverList.cpp

```diff
@@ -0,0 +1,144 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "nsObserverList.h"
+
+#include "nsAutoPtr.h"
+#include "nsCOMArray.h"
+#include "nsISimpleEnumerator.h"
+#ifndef GECKO_MEDIA_CRATE
+#include "xpcpublic.h"
```

Comment:
If this is the only change, and we build with this change, then we should able to upstream this change and move this file out of glue.

Reply:
Agreed :)